kratos/geometries/geometry.h

```diff
@@ -554,26 +559,44 @@ class Geometry
             *i = typename PointType::Pointer( new PointType( **i ) );
     }
 
-     // virtual Kratos::shared_ptr< Geometry< Point > > Clone() const
-     // {
-     //     Geometry< Point >::PointsArrayType NewPoints;
+    ///@}
+    ///@name Id
+    ///@{
 
-     //     //making a copy of the nodes TO POINTS (not Nodes!!!)
+    /// Id of this Geometry
+    IndexType Id() const
+    {
+        return mId;
+    }
+
+    /// Sets Id of this Geometry
+    void SetId(IndexType Id)
+    {
+        KRATOS_ERROR_IF(IsSelfAssignedId(Id) && IsGeometryIdString(Id))
+            << "Id out of range. The first bit of the Id is used to "
```

Comment:
I would only suggest that the id should be less than 2**62 which is the practical info for who writing a mesh